src/vellum/workflows/nodes/displayable/bases/inline_prompt_node/node.py

```diff
@@ -134,6 +135,14 @@ def _get_prompt_event_stream(self) -> Iterator[AdHocExecutePromptEvent]:
             )
 
     def _process_prompt_event_stream(self) -> Generator[BaseOutput, None, Optional[List[PromptOutput]]]:
+        try:
+            self.blocks = compile_blocks_to_prompt_blocks(self.blocks)  # type: ignore
```

Comment:
I think this could be as simple as:

`self.blocks = [PromptBlock.model_validate(block) block in blocks]`

Reply:
Actually, `PromptBlock` is a union, so you may need to do something like

```python
exec_config = PromptExecConfig.model_validate({
    "blocks": self.blocks,
    ..., # other fields here too
})

self.blocks = exec_config.blocks # or just feed `exec_config` directly into the eventual api request to vellum
```